temporian/core/graph.py

```diff
@@ -161,6 +163,44 @@ def p3(title, l):
             p3("Output", self.outputs)
         return s
 
+    def apply_on_inputs(
+        self, named_inputs: Dict[str, EventSetNode]
+    ) -> Dict[str, EventSetNode]:
+        """Applies the operators in this graph to new inputs.
+
+        Note that the objects in the modified graph are very inconsistent, but
+        that's okay since we won't use it anymore. When running it or save it
+        the graph will be re-inferred.
+
+        Args:
+            named_inputs: The new inputs to the graph.
+
+        Returns:
+            The graph's named outputs.
+        """
+
+        # Avoid messing with the objects in the received graph
+        # Else each time we apply the graph on inputs it changes
+        # Might be possible to optimize by copying only what's necessary
+        g = deepcopy(self)
+
+        assert g.named_inputs is not None
+        assert g.named_outputs is not None
+
+        for name, new_node in named_inputs.items():
+            if name not in g.named_inputs:
+                raise ValueError(
+                    f"Input node {name} is not in the graph's inputs. Inputs:"
+                    f" {g.named_inputs}"
+                )
+            old_node = g.named_inputs[name]
+            # Replace node as input in all operators that depend on it
+            for operator in g.operators:
```

Comment:
Can you add a TODO to create and manage a mapping from named_input to operator in the graph to maje this method more efficient?